src/locales/localeConfiguration.ts

```diff
@@ -0,0 +1,11 @@
+import { addLocaleData } from "react-intl";
+import * as en from "react-intl/locale-data/en";
+import localeData from "./data";
+
+export const getLocalizedResources = () => {
+    addLocaleData([...en]);
```

Comment:
You forgot to add fr support (your json resource already contains french translations)
const language = (navigator.languages && navigator.languages[0]) ||
                  navigator.language ||
                  navigator.userLanguage;

const languageWithoutRegionCode = language.toLowerCase().split(/[_-]+/)[0];

const messages = localeData[languageWithoutRegionCode] || localeData[language] || localeData.en;

this is the idea...